### PR TITLE
fix: query filter in list task endpoint

### LINKED ISSUE #175: fix: Getting list of tasks not working without name prefix
**Describe the bug**
Making a GET request to the `/tasks` endpoint proTES with an empty body always returns an empty list of tasks, even when a task has been created. This is because of [this](https://github.com/elixir-cloud-aai/proTES/blob/dev/pro_tes/ga4gh/tes/task_runs.py#L282) line in `task_runs.py`. Without a name prefix, `str(kwargs.get("name_prefix"))` returns the string `"None"` meaning `name_prefix is not None` is always True. As a result, a filter of `"^None"` is applied when querying the db.

**To Reproduce**
1. Run proTES
2. Create a task (POST to /tasks with valid body (e.g. `{"executors": [{"image": "alpine", "command": ["echo", "hello"]}]}`)
3. Get list of tasks (GET to /tasks with empty body)
4. Task list is empty

**Expected behavior**
The task list should contain the newly created task ID and status.

**Screenshots**
POST request made successfully:
Screenshot 2024-05-20 at 2 15 27 PM

GET request returns empty task list:
Screenshot 2024-05-20 at 2 16 01 PM

### LINKED ISSUE #177: fix: Handle None Values Correctly in List Task Endpoint Query Filter
When building the query filter with potential None values for name_prefix and user_id, the filter should be adjusted accordingly to avoid incorrect queries.

Expected Behavior:
- If name_prefix is None, filter_dict should be empty ({}).
- If user_id is None, it should not be added to the filter.

Actual Behavior:
Currently, if name_prefix is None, the filter becomes:
`filter_dict = {'task_original.name': {'$regex': '^None'}}`

while in the case of None Type the filter_dict should be empty like` filter_dict = {}`
The actual issue is here:
```
        name_prefix: str = str(kwargs.get("name_prefix"))

        if name_prefix is not None:
            filter_dict["task_original.name"] = {"$regex": f"^{name_prefix}"}
```
and,
`filter_dict["user_id"] = kwargs.get("user_id")`

### PR DESCRIPTION
**Pull request by Ayush5120**
Fixes #177
Fixes #175 

## Summary by Sourcery

This pull request addresses a bug in the list tasks endpoint by ensuring that 'user_id' and 'name_prefix' filters are only added to the filter dictionary if they are not None.

* **Bug Fixes**:
    - Fixed an issue where the 'user_id' and 'name_prefix' filters in the list tasks endpoint were not properly handling None values.

### CODE REVIEW

**Review by uniqueg — COMMENTED** [review]
Looks good to me, but before merging: @athith-g - can you please confirm that this PR solves #175 for you?

**Review by athith-g — APPROVED** [review]
It does fix #175 👍

**Review by uniqueg — CHANGES_REQUESTED** [review]
That's good to hear. However, linting and tests fail - please check, @Ayush5120 :pray:

File: pro_tes/utils/db.py [hunk 1]
```diff
@@ -23,17 +23,17 @@ class DbDocumentConnector:
     """
 
     def __init__(
-        self,
-        collection: Collection,
-        worker_id: str,
+            self,
+            collection: Collection,  # type: ignore
+            worker_id: str,
```
Comment (uniqueg) [inline pro_tes/utils/db.py]: Why the double indenting (8 chars) on these? Did `black` do that? We should have only one level of indentation for the function signatures

File: pro_tes/utils/db.py [hunk 2]
```diff
@@ -23,17 +23,17 @@ class DbDocumentConnector:
...
     ) -> None:
         """Construct object instance."""
-        self.collection: Collection = collection
+        self.collection: Collection = collection  # type: ignore
```
Comment (uniqueg) [inline pro_tes/utils/db.py]: Why the `type: ignore`?

File: pro_tes/utils/db.py [hunk 3]
```diff
@@ -51,7 +51,7 @@ def get_document(
         """
         if projection is None:
             projection = {"_id": False}
-        document_unvalidated = self.collection.find_one(
+        document_unvalidated = self.collection.find_one(  # type: ignore
```
Comment (uniqueg) [inline pro_tes/utils/db.py]: And here?

**Review by uniqueg — COMMENTED** [review]
Can you please explain why we are ignoring types?

File: pro_tes/utils/db.py (hunk 1 above)
Comment (Ayush5120) [reply pro_tes/utils/db.py]: my pycharm by default reformats to 8 chars indent, let me fix that in the next commit.

File: pro_tes/utils/db.py (hunk 2 above)
Comment (Ayush5120) [reply pro_tes/utils/db.py]: After removing the deprecated insert, this type: ignore is now not needed. I forgot to remove it, fixed it in the next commit

File: pro_tes/utils/db.py [hunk 4]
```diff
@@ -106,17 +104,18 @@ def upsert_fields_in_root_object(
...
+                {"worker_id": self.worker_id},
+                {
+                    "$set": {
+                        ".".join([root, key]): value
+                        for (key, value) in kwargs.items()
+                    }
+                },
+                projection=projection,
+                return_document=ReturnDocument.AFTER,
+            ))
```
Comment (uniqueg) [inline pro_tes/utils/db.py]: Was this formatted by `black`? If not, I think the second closing parenthesis should be on a separate line with less indent.

**Review by uniqueg — CHANGES_REQUESTED** [review]

File: pro_tes/utils/db.py (hunk 4 above)
Comment (Ayush5120) [reply pro_tes/utils/db.py]: Fixed the indent

**Review by uniqueg — APPROVED** [review]

**Pull request merged**